examples/expressjs-ethr/src/identity.ts

```diff
@@ -0,0 +1,39 @@
+import * as Daf from 'daf-core'
+import { core } from './framework'
+
+export const getIdentity = async () => {
+  // Get of create new identity
+  let identity: Daf.AbstractIdentity
+  const identities = await core.identityManager.getIdentities()
+  if (identities.length > 0) {
+    identity = identities[0]
+  } else {
+    const identityProviders = await core.identityManager.getIdentityProviderTypes()
+    identity = await core.identityManager.createIdentity(identityProviders[0].type)
+  }
+
+  return identity
+}
+
+export const setServiceEndpoint = async (identity: Daf.AbstractIdentity, serviceEndpoint: string) => {
+  // Check if DID Document contains current serviceEndpoint
+  const didDoc = await core.didResolver.resolve(identity.did)
+  const exists = didDoc?.service?.find(item => item.serviceEndpoint === serviceEndpoint)
+  if (!exists) {
+    try {
+      const txHash = await identity.identityController.addService({
```

Comment:
@mirceanis @simonas-notcat one question, are you guys baking in tx cost into this, or where are you setting the possibility to cover the cost for this tx?

Reply:
While there are some defaults, TX cost is ultimately controlled by the web3 provider.

Worth noting that this is an example and not necessarily meant to be used as a general case.

Most likely the general case involves sharing a callback URL along with the the request, such that updating the DID doc is not a requirement.
This is an example that shows did-doc based messaging is possible.

Reply:
@mirceanis yeah I see. Not I just like the easy way of updating the did doc, compared to the necessities today. We want to set a metadata service on a did doc. And cover that cost. But do you have an example of where you setup the web3 provider for that?

Reply:
Not yet, but feel free to contribute one ;)

Reply:
@mirceanis hehe im will try when I dig deeper in this. But do you have any pointers to where you can send in the web3? So I just have start basis :)